tools/ResetConcurrencyFlag/README.md

```diff
@@ -0,0 +1,38 @@
+# Retries concurrency flag reset
+
+An [issue in ServiceControl 1.6.0 to 1.6.3](https://github.com/Particular/ServiceControl/pull/565) was identified that causes the retries feature of ServiceControl to stall and not process any subsequent retries. This issue is fixed in [ServiceControl 1.6.4](https://github.com/Particular/ServiceControl/releases/tag/1.6.4), however in order to reset stalled retries a manual step is needs to be performed by the user, this is where this tool will help.
+
+This tool resets the concurrecy flag in stalled messages that were part of a batch retry.
+
+## How to identify messages that are stalled?
+
+"Stalled" messages are messages that have been retried but failed to return to the input queue of an endpoint.
+To identify these messages, you need to open the `particular.servicecontrol.errors` queue (note the queue name changes based on the name of the windows service) and look at the headers of the current messages in there.
+If you find messages with `ServiceControl.Retry.UniqueMessageId` header, eg
+```xml
+<HeaderInfo>
+	<Key>ServiceControl.Retry.UniqueMessageId</Key>
+	<Value>efb8fb3d-7649-e0ea-f3fc-f77fc79abc3b</Value>
+</HeaderInfo>
+```
+You need to copy the value of that header and run the tool against that value, eg
+```cmd
+ResetMessageRetry.exe efb8fb3d-7649-e0ea-f3fc-f77fc79abc3b
+```
+
+This should give the following output:
+```txt
+Resetting message...
+Done
+```
+
+**Note:**
+The tool assumes that RavenDB instance is exposed at the default url, i.e. http://localhost:33333/storage. If you have altered the URL in the ServiceControl configuration, you can pass the full URL to the executable like this:
```

Comment:
Can we link this to the doco explaining how to expose raven (if it isn't already) http://docs.particular.net/servicecontrol/use-ravendb-studio